### ISSUE TITLE
App crashes when trying to report a native objc exception

### ISSUE THREAD

**Issue by Therzok** [mono/monodevelop#7638]
Trying to show an alert dialog while an NSException is in flight will cause the app to crash:
http://source.monodevelop.com/#MonoDevelop.Ide/MonoDevelop.Ide/LogReportingStartup.cs,45

```
ERROR [2019-05-22 23:12:38Z]: Unhandled ObjC exception
MonoDevelop.MacIntegration.MacPlatformService+MarshalledObjCException: NSGenericException: -[NSApplication runModalForWindow:] may not be invoked inside of transaction begin/commit pair, or inside of transaction commit (usually this means it was invoked inside of a view's -drawRect: method.)
2019-05-22 23:12:38.498 VisualStudio[1536:25618] *** Terminating app due to uncaught exception 'NSGenericException', reason: '-[NSApplication runModalForWindow:] may not be invoked inside of transaction begin/commit pair, or inside of transaction commit (usually this means it was invoked inside of a view's -drawRect: method.)'
```

```
FATAL ERROR [2019-05-22 23:12:38Z]: Visual Studio failed to start. Some of the assemblies required to run Visual Studio (for example gtk-sharp)may not be properly installed in the GAC.
System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'FormattedLine'.
  at Microsoft.VisualStudio.Text.Formatting.Implementation.CocoaFormattedLine.ThrowIfInvalid () [0x00013] in /Users/vsts/agent/2.150.3/work/1/s/md-addins/external/vs-editor-core/src/Editor/Text/Impl/CocoaView/FormattedTextSource/CocoaFormattedLine.cs:193 
  at Microsoft.VisualStudio.Text.Formatting.Implementation.CocoaFormattedLine.get_Baseline () [0x00000] in /Users/vsts/agent/2.150.3/work/1/s/md-addins/external/vs-editor-core/src/Editor/Text/Impl/CocoaView/FormattedTextSource/CocoaFormattedLine.cs:886 
  at Microsoft.VisualStudio.Text.Editor.Implementation.CocoaTextViewControl.get_FirstBaselineOffsetFromTop () [0x0001a] in /Users/vsts/agent/2.150.3/work/1/s/md-addins/external/vs-editor-core/src/Editor/Text/Impl/CocoaView/WpfViewImpl/CocoaTextViewControl.cs:253 
  at (wrapper managed-to-native) Gtk.Application.gtk_main()
  at Gtk.Application.Run () [0x00001] in /Users/builder/jenkins/workspace/build-package-osx-mono-pullrequest/pr/external/bockbuild/builds/gtk-sharp-None/gtk/Application.cs:145 
  at MonoDevelop.Ide.IdeStartup.Run (MonoDevelop.Ide.MonoDevelopOptions options) [0x00429] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs:216 
  at MonoDevelop.Ide.IdeStartup.Main (System.String[] args, MonoDevelop.Ide.Extensions.IdeCustomizer customizer) [0x0009e] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs:716 
```

> VS bug [#897638](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/897638)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.1.0.2541 (release-8.1)

Author: therzok
Commit: 6a2824406c8d754333d2b6502095e182fbec26fe (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.3.0.4 (master)

Author: therzok
Commit: 43d5d22d42d76204aa66571afbeb247a166af2ec (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.2.0.542 (release-8.2)

Author: therzok
Commit: 43d5d22d42d76204aa66571afbeb247a166af2ec (mono/monodevelop)
